src/main/java/com/nimbusds/jose/crypto/ECDH1PUEncrypter.java

```diff
@@ -0,0 +1,271 @@
+     *
+     * @param publicKey The public EC key. Must not be {@code null}.
+     *
+     * @throws JOSEException If the elliptic curve is not supported.
+     */
+    public ECDH1PUEncrypter(final ECPrivateKey privateKey, final ECPublicKey publicKey)
+        throws JOSEException {
+
+        this(privateKey, publicKey, null);
+    }
+
+
+    /**
+     * Creates a new Elliptic Curve Diffie-Hellman encrypter with an
+     * optionally specified content encryption key (CEK).
+     *
+     * @param privateKey            The private EC key. Must not be
+     *               			   {@code null}.
+     *
+     * @param publicKey            The public EC key. Must not be
+     *                             {@code null}.
+     * @param contentEncryptionKey The content encryption key (CEK) to use.
+     *                             If specified its algorithm must be "AES"
+     *                             and its length must match the expected
+     *                             for the JWE encryption method ("enc").
+     *                             If {@code null} a CEK will be generated
+     *                             for each JWE.
+     * @throws JOSEException       If the elliptic curve is not supported.
+     */
+    public ECDH1PUEncrypter(final ECPrivateKey privateKey,
+                            final ECPublicKey publicKey,
+                            final SecretKey contentEncryptionKey)
+            throws JOSEException {
+
+        super(Curve.forECParameterSpec(publicKey.getParams()));
+
+        this.privateKey = privateKey;
+        this.publicKey = publicKey;
+
+        if (contentEncryptionKey != null && (contentEncryptionKey.getAlgorithm() == null || !contentEncryptionKey.getAlgorithm().equals("AES")))
+            throw new IllegalArgumentException("The algorithm of the content encryption key (CEK) must be AES");
+
+        this.contentEncryptionKey = contentEncryptionKey;
+    }
+
+
+    /**
+     * Returns the public EC key.
+     *
+     * @return The public EC key.
+     */
+    public ECPublicKey getPublicKey() {
+
+        return publicKey;
+    }
+
+
+    /**
+     * Returns the private EC key.
+     *
+     * @return The private EC key.
+     */
+    public ECPrivateKey getPrivateKey() {
+
+        return privateKey;
+    }
+
+
+    @Override
+    public Set<Curve> supportedEllipticCurves() {
+
+        return SUPPORTED_ELLIPTIC_CURVES;
+    }
+
+
+    @Override
+    public JWECryptoParts encrypt(final JWEHeader header, final byte[] clearText)
+        throws JOSEException {
+
+        ECDH1PU.validateSameCurve(privateKey, publicKey);
+
+        // Generate ephemeral EC key pair on the same curve as the consumer's public key
+        KeyPair ephemeralKeyPair = generateEphemeralKeyPair(publicKey.getParams());
+        ECPublicKey ephemeralPublicKey = (ECPublicKey)ephemeralKeyPair.getPublic();
+        ECPrivateKey ephemeralPrivateKey = (ECPrivateKey)ephemeralKeyPair.getPrivate();
```

Comment:
Since nimbus jose does not support JSON serialization, this case has not been solved yet. During implementing json serialization I ran into this issue, But I think it should be corrected in a separate PR. in my opinion we should make big contribute in API nimbus jose becouse of multi recipients

Reply:
I'm not sure I understand, JSON serialization seems unrelated to being able to re-use the ephemeral key in the code if we encrypt for multiple recipients, no?

Reply:
Currently Nimbus doesn't support multiple recipients (using the same ephemeral keys for multiple recipients).
That's one of the things we are going to implement and contribute. 
We suggest to do it in a separate PR, as it will require changes in the core Nimbus interfaces. I think it's better to contribute step-by-step instead of sending a huge PR with a lot of changes (including changes to existing Nimbus interfaces).

Reply:
Still, what I'm saying isn't requiring changes on the Nimbus side:
we are creating an ephemeral key in the method, but we could also have an ephemeral key be part of the ECDH1PUEncrypter class as an attribute, and if it exist use it, otherwise set it to a new key.

Maybe add an explicit counter to the ECDH1PUEncrypter class like "reuseEpk" that we can decrease manually in `encrypt` and we could reset the epk attribute back to null after so many encryptions... 
So a codebase willing to encrypt for multiple recipients with the same EPK would need to set it to the number of recipients before calling encrypt.

That doesn't require changing Nimbus API. 
And it would also allow for concurrent encryptions for all recipients by just setting the EPK for each of them to the one we want to use.

(But would such an API be acceptable for us @CluEleSsUK ?)

Reply:
1) The same changes for EPK will need to be done ECDH-ES which is part of existing Nimbus code. So, we will have to change Nimbus core code to support it.
2) Please note, that these changes and API will not be used by end users. This is not an API that we exposer. It will be used by DID Comm lib only under the hood.
3) I suggest to follow step-by-step principle and our plans from the project planning point of view. We do plan to support the feature you are talking about, but let's do it after we finish the basic ECDH-1PU support and (hopefully) send the first PR to Nimbus.

Reply:
Okay, sounds good as long as multiple recipients using the same EPK is supported at the end.